### ISSUE TITLE
initial  large tf.variable error

### ISSUE THREAD

**Issue by B3N50N** [tensorflow/tensorflow#17941]
### System information
- **Have I written custom code: yes 
- **OS Platform and Distribution Ubuntu 16.04**:
- **TensorFlow installed from source **:
- **TensorFlow version 1.6**:
- **Python version 3.5**: 
- **Bazel version 0.11.1**:
- **GCC/Compiler version (5)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:
### Describe the problem
when i'm using same code, only the weight variable have different size
and when weight array is small everything is fine, but when too big it show shape issue

### Source code / logs
souce code:
with tf.device('/cpu:0'):
	W_fc1 = weight_variable([32768, 35000],"W_fc1")#32768
	b_fc1 = bias_variable([35000],"b_fc1")
	h_pool5_flat = tf.reshape(h_pool5, [-1, 16*16*128])
	h_fc1 = tf.nn.relu(tf.matmul(h_pool5_flat, W_fc1) + b_fc1,name="h_fc1")
	h_fc1_drop = tf.clip_by_value(tf.cast(tf.nn.dropout(tf.cast(h_fc1,tf.float32), keep_prob),tf.float16, name="h_fc1_drop"),min_clip,max_clip, name="h_fc1_drop")
	h_fc1_drop_1 = tf.where(tf.is_nan(h_fc1_drop), tf.constant(min_clip,dtype=tf.float16, shape=h_fc1_drop.shape), h_fc1_drop)

##down sampleing fc2 layer##
#with tf.device('/cpu:0'):
	W_fc2 = weight_variable([35000,65536],"W_fc2")
	b_fc2 = bias_variable([65536],"b_fc2")
	h_fc2 = tf.clip_by_value(tf.nn.relu(tf.matmul(h_fc1_drop_1, W_fc2) + b_fc2),min_clip,max_clip, name="h_fc2")
	h_fc2_1 = tf.where(tf.is_nan(h_fc2), tf.constant(min_clip,dtype=tf.float16, shape=h_fc2.shape), h_fc2)
..........
loss = tf.losses.mean_squared_error(y_reshape,prediction_1)
train_step = tf.train.AdamOptimizer(lr).minimize(loss)

when W_fc1 is as big as the above 
it show the error
   File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3271, in create_op
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1650, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [35000,65536] rhs shape= [32768,35000]
         [[Node: W_fc2/Adam_1/Assign = Assign[T=DT_HALF, _class=["loc:@W_fc2"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](W_fc2/Adam_1, W_fc2/Adam/Initializer/zeros)]]

**Comment by poxvoculi** [comment]
The error suggests that the shapes of W_fc1 and W_fc2 need to be commensurate.  When you set W_fc1 to a smaller size, did W_fc2 also have a smaller size, or was it [35000,65536]?

**Comment by B3N50N** [comment]
@poxvoculi yes i do , i'm sure all the setting are correct ,if not it'll crash at the begin , but the error comes out while the sess.run(init)

**Comment by poxvoculi** [comment]
I'm afraid I don't understand.  Can you post the full program that works correctly, and also the full program that fails?

**Comment by B3N50N** [comment]
well the only different part of success and fail vertion are the size of the wieght 
and i'm sure i did it correct but i can still post my code here
https://gist.github.com/B3N50N/aafd02d4c70cc7d24ce45c5fe472555f

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by poxvoculi** [comment]
I'm sorry but there's not enough information here for me to work with, in a time efficient fashion.  It sounds to me like some kind of usage problem.  Usage questions are best asked on stack overflow.  If you believe there is a TensorFlow bug, please try to post a very small example that demonstrates the problem.
